### PR TITLE
Bugfix validate https redirect url

### PR DESCRIPTION
**Pull request by simen-rekkedal**
## Related Issue(s)
Bug report:
- #849

## Verification
- [x] **Your** code builds clean without any errors or warnings
- [x] Manual testing done (required)
- [x] Relevant automated test added (if you find this hard, leave it and we'll help out)
- [x] All tests run green

## Documentation
- [ ] User documentation is updated with a separate linked PR in [altinn-studio-docs.](https://github.com/Altinn/altinn-studio-docs) (if applicable)

### CODE REVIEW

File: src/Authentication/Helpers/AuthenticationHelper.cs [hunk 1]
```diff
@@ -20,6 +20,11 @@ namespace Altinn.Platform.Authentication.Helpers
     /// </summary>
     public static class AuthenticationHelper
     {
+        /// <summary>
+        /// Regular expression for valid uri
+        /// </summary>
+        public static readonly Regex ValidUri = new Regex(@"^https://[a-zA-Z0-9.-]+.[a-zA-Z]{2,}$");
```
Comment (Alxandr) [inline src/Authentication/Helpers/AuthenticationHelper.cs:26]: This isn't used? And if it is, should be compiled.

File: src/Authentication/Helpers/AuthenticationHelper.cs [hunk 2]
```diff
@@ -380,5 +382,11 @@ public static int GetUserId(HttpContext context)
 
             return 0;
         }
+
+        private static bool IsValidUriWithHttps(string uri)
+        {
+            return Uri.TryCreate(uri, UriKind.Absolute, out Uri? validatedUri)
```
Comment (Alxandr) [inline src/Authentication/Helpers/AuthenticationHelper.cs:388]: Don't parse a uri that's already been parsed.

**Review by Alxandr — APPROVED** [review]

File: src/Authentication/Helpers/AuthenticationHelper.cs (hunk 1 above)
Comment (simen-rekkedal) [reply src/Authentication/Helpers/AuthenticationHelper.cs]: removed

File: src/Authentication/Helpers/AuthenticationHelper.cs (hunk 2 above)
Comment (simen-rekkedal) [reply src/Authentication/Helpers/AuthenticationHelper.cs]: ok

**Pull request merged**